js/piwik.js

```diff
@@ -592,6 +592,19 @@ if (typeof window.Matomo !== 'object') {
 
             return matches ? matches[1] : url;
         }
+        function queryStringify(data, filterFn, throwErrorIfNotFilterFn) {
+            var queryString = '', k, filter = filterFn || function(value) { return value; };
+            for (k in data) {
+                if (data.hasOwnProperty(k)) {
+                    if (filter(data[k])) {
+                        queryString += '&' + encodeWrapper(k) + '=' + encodeWrapper(data[k]);
+                    } else if (throwErrorIfNotFilterFn && undefined !== data[k]) {
+                        throw new Error('Parameter value not allowed');
```

Comment:
BTW here users wouldn't know which parameter value is not allowed and which value they provided. In these cases we say something like `Parameter "pf_dm1" provided value "sde3" is not valid. Please provided a numeric value.` This will help them to know which parameter is wrong, what they had provided and how they can fix it.